test/envoye2e/env/setup.go

```diff
@@ -370,6 +381,21 @@ func (s *TestSetup) WaitServerEnvoyReady() {
 	s.WaitEnvoyReady(s.Ports().ServerAdminPort)
 }
 
+func checkStats(expectedStatsMap map[string]int, actualStatsMap map[string]int) error {
+
+	for eStatsName, eStatsValue := range expectedStatsMap {
+		aStatsValue, ok := actualStatsMap[eStatsName]
+		if !ok && eStatsValue != 0 {
```

Comment:
Why it is ok to not find the stat in actual map if expected value is 0? For gauge stat, this seems to be wrong.

Reply:
yeah removed that logic... it was carried over from existing envoy stats verification function..